### PR TITLE
Expose bicubic mode for torch::nn::functional::grid_sample in LibTorch

### PR DESCRIPTION
**Pull request by inventshah**
When bicubic interpolation was added to grid_sampler in #44780, `GridSampleFuncOptions` was not updated to allow a user to use bicubic mode in LibTorch, even though the function could handle it. This PR fixes the parity such that LibTorch's  `torch::nn::functional::grid_sample` behaves the same as PyTorch's `torch.nn.functional.grid_sample`.

Existing users can directly use `torch::grid_sampler` but must know what int to pass for the interpolation (2 for bicubic) and padding mode parameters, which is not ideal.

### CODE REVIEW

**Review by Skylion007 — APPROVED** [review]

**Comment by cyyever** [comment]
@inventshah  Your linting fixes shouldn't include third-party libraries.

**Comment by inventshah** [comment]
@cyyever Sorry! Should be fixed now.

**Comment by inventshah** [comment]
Hi @cyyever, at a glance the test failure does not appear to be related to the PR, since `TextDecoderTest::test_llama3_2_text_decoder_aoti` passed in the [pre-lint fix CI job](https://github.com/pytorch/pytorch/actions/runs/14482157125/job/40622368075#step:22:9612), and also does not use `grid_sample` at least within the first few layers of dependencies. Thoughts?

 I'll also start work on replicating the executorch tests locally.
